moveit_ros/visualization/handeye_calibration_rviz_plugin/src/handeye_control_widget.cpp

```diff
@@ -0,0 +1,828 @@
+{
+  for (const std::string& plugin : plugins)
+    if (!plugin.empty() && createSolverInstance(plugin))
+    {
+      const std::vector<std::string>& solvers = solver_->getSolverNames();
+      for (const std::string& solver : solvers)
+      {
+        std::string solver_name = plugin + "/" + solver;  // solver name format is "plugin_name/solver_name"
+        calibration_solver_->addItem(tr(solver_name.c_str()));
+      }
+    }
+}
+
+std::string ControlTabWidget::parseSolverName(const std::string& solver_name, char delimiter)
+{
+  std::vector<std::string> tokens;
+  std::string token;
+  std::istringstream tokenStream(solver_name);
+  while (std::getline(tokenStream, token, delimiter))
+  {
+    tokens.push_back(token);
+  }
+  return tokens.back();
+}
+
+bool ControlTabWidget::takeTranformSamples()
+{
+  // Store the pair of two tf transforms and calculate camera_robot pose
+  try
+  {
+    geometry_msgs::TransformStamped cTo;
+    geometry_msgs::TransformStamped bTe;
+
+    // Get the transform of the object w.r.t the camera
+    cTo = tf_buffer_->lookupTransform(frame_names_["sensor"], frame_names_["object"], ros::Time(0));
+
+    // Get the transform of the end-effector w.r.t the robot base
+    bTe = tf_buffer_->lookupTransform(frame_names_["base"], frame_names_["eef"], ros::Time(0));
+
+    // save the pose samples
+    effector_wrt_world_.push_back(tf2::transformToEigen(bTe));
+    object_wrt_sensor_.push_back(tf2::transformToEigen(cTo));
+
+    ControlTabWidget::addPoseSampleToTreeView(cTo, bTe, effector_wrt_world_.size());
+  }
+  catch (tf2::TransformException& e)
+  {
+    ROS_WARN("TF exception: %s", e.what());
+    return false;
+  }
+
+  return true;
+}
+
+bool ControlTabWidget::solveCameraRobotPose()
+{
+  if (solver_ && !calibration_solver_->currentText().isEmpty())
+  {
+    bool res = solver_->solve(effector_wrt_world_, object_wrt_sensor_, sensor_mount_type_,
+                              parseSolverName(calibration_solver_->currentText().toStdString(), '/'));
+    if (res)
+    {
+      camera_robot_pose_ = solver_->getCameraRobotPose();
+
+      // Update camera pose guess in context tab
+      Eigen::Vector3d t = camera_robot_pose_.translation();
+      Eigen::Vector3d r = camera_robot_pose_.rotation().eulerAngles(0, 1, 2);
+      Q_EMIT sensorPoseUpdate(t[0], t[1], t[2], r[0], r[1], r[2]);
+
+      // Publish camera pose tf
+      std::string& from_frame = frame_names_[from_frame_tag_];
+      std::string& to_frame = frame_names_["sensor"];
+      if (!from_frame.empty() && !to_frame.empty())
+      {
+        tf_tools_->clearAllTransforms();
+        return tf_tools_->publishTransform(camera_robot_pose_, from_frame, to_frame);
+      }
+      else
+      {
+        ROS_ERROR_STREAM_NAMED(LOGNAME, "Invalid key used for reading the frame names.");
+        return false;
+      }
+    }
+  }
+  else
+  {
+    ROS_ERROR_STREAM_NAMED(LOGNAME, "No available handeye calibration solver instance.");
```

Comment:
I don't know if is this line that should be changed, but currently the solving pipeline fails silently, and the user does not know if there is an issue. At some point there should give an error popup instead of just a silent error stream.